### ISSUE TITLE
sidebar - on navigation highlight current item

### ISSUE THREAD

**Issue by kuchaguangjie** [themesberg/flowbite-svelte#280]
Similar as this one: https://sveltematerialui.com/demo/drawer/

**Comment by shinokada** [comment]
I think this is what you need, https://flowbite-svelte.com/components/sidebar#Active_item

Can you update to the latest version, v0.26.17?

```
// watch out! This will update all dependencies
npm update
// if you want to update only Flowbite-Svelte
npm i -D flowbite-svelte@latest
```

And let me know.

**Comment by kuchaguangjie** [comment]
Thanks.
I think the variable should be:
> $: activeUrl = $page.url.pathname
other than that , it works.

**Comment by kuchaguangjie** [comment]
Another enhancement might be, what if the url got some query params after `?` or some anchor `#`, the `activeUrl` should probably exclude that, in most case, or it depends on the use case.

But I think the programmer can do it, though it would be nice to mention that, and give some example of how to define the `activeUrl` variable, to include / exclude `?` and/or `#` parts.

**Comment by kuchaguangjie** [comment]
`<SidebarDropdownItem>` doesn't support the `active` property yet.

**Comment by kuchaguangjie** [comment]
When item already active, on:hover, it better not change color maybe, this is miner though.

**Comment by shinokada** [comment]
Can you try this if it picks up `#` and `?`?

```
<script>
    import { onMount } from 'svelte';
    let url = ``;
    onMount(() => url = window.location.href);
</script>

{url}
```

**Comment by kuchaguangjie** [comment]
In my test:
- `'/'+(location.pathname+location.search).substr(1)` would pick up `?` part, but not `#` part.

but both `window` and `location` requires to be within `onMount()`, the mount event only happen once, unless refresh by hand.
when click sidebar items, won't trigger the event, so probably this won't work anyway.

**Comment by shinokada** [comment]
If you have a good solution, please let me know. I will include it in the doc.

**Comment by kuchaguangjie** [comment]
>   $: activeUrl = $page.url.pathname + $page.url.hash
This works in my test.
If the url is `http://localhost:3000/home#about?a=1&b=2`, then the `hash` will be `#about?a=1&b=2`

BTW, probably need change `const activeUrl` to `$: activeUrl` in the document.

**Comment by shinokada** [comment]
Yes, I will change it.

**Comment by kuchaguangjie** [comment]
![hash](https://user-images.githubusercontent.com/4230959/189606343-8d71569f-8457-4f94-9de6-88e1d128f81a.jpg)

**Comment by kuchaguangjie** [comment]
And, the user can decide whether include the `hash` part or not, since the user themselves will define the expression for `activeUrl` and `<SidebarItem active={}>`, this is pretty flexible.

**Comment by shinokada** [comment]
Yes, you are right. It's not UI's responsibility and uses should define them.
We just want to give an example/starter what you can do.

**Comment by kuchaguangjie** [comment]
Seemss without `hash`, seems the `?` part is in `search`.
![search](https://user-images.githubusercontent.com/4230959/189608138-82b41acd-aab7-4bf0-80cf-1af8190814ab.jpg)

So, the user needs to decide how to use `hash` and `search` according to their own `href` in `<SidebarItem href="...">`.

**Comment by kuchaguangjie** [comment]
> `<SidebarDropdownItem>` doesn't support the `active` property yet.

BTW is this feature also planned to do ?

**Comment by shinokada** [comment]
I updated and add an example, but you need to add your own logic for `containPath` function. See https://flowbite-svelte.com/components/sidebar#Active_item.

If you have an idea for this function, let me know.

**Comment by kuchaguangjie** [comment]
I tried the example, it seems not working, maybe need a new tag ?

**Comment by kuchaguangjie** [comment]
```
      <Sidebar>
        <SidebarWrapper>
          <SidebarGroup>
            <SidebarDropdownWrapper label="Home" isOpen={true}>
              <SidebarDropdownItem label="Post" href='/home/post' active={activeUrl === '/home/post'}/>
              <SidebarDropdownItem label="Quiz" href='/home/quiz' active={activeUrl === '/home/quiz'}/>
            </SidebarDropdownWrapper>
          </SidebarGroup>
        </SidebarWrapper>
      </Sidebar>
```
version: `"flowbite-svelte": "^0.26.18",`

**Comment by shinokada** [comment]
Can you try the second example of https://flowbite-svelte.com/components/sidebar#Active_item ?

**Comment by kuchaguangjie** [comment]
Seems there is another small issue, the arrow on `SidebarDropdownWrapper` doesn't change when open or close the drop-down list:
![arrow](https://user-images.githubusercontent.com/4230959/191001460-1bb4400a-0f28-4082-ae66-03745f76d91b.jpg)

**Comment by kuchaguangjie** [comment]
Another small issue:
When set `isOpen="false"` on `<SidebarDropdownWrapper>`, it will still open; removing the `isOpen` property it will close.

**Comment by shinokada** [comment]
Can you open a new issue?
Thanks.

**Comment by kuchaguangjie** [comment]
https://github.com/themesberg/flowbite-svelte/issues/302
